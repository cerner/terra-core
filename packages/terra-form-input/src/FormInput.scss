:local {
  .form-input {
    background-color: var(--terra-form-input-background-color, #fff);
    background-image: var(--terra-form-input-focus-animation-background, none);
    border-color: var(--terra-form-input-border-color, #4a4a4a);
    border-radius: var(--terra-form-input-border-radius, 0.2rem);
    border-style: var(--terra-form-input-border-style, solid);
    border-width: var(--terra-form-input-border-width, 1px);
    box-shadow: var(--terra-form-input-box-shadow, none);
    color: var(--terra-form-input-color, #000);
    font-family: inherit;
    font-size: var(--terra-form-input-font-size, 1.143rem);
    line-height: var(--terra-form-input-line-height, 1.143);
    opacity: var(--terra-form-input-opacity, 1);
    padding-bottom: var(--terra-form-input-padding-bottom, 0.571rem);
    padding-left: var(--terra-form-input-padding-left, 0.357rem);
    padding-right: var(--terra-form-input-padding-right, 0.357rem);
    padding-top: var(--terra-form-input-padding-top, 0.571rem);
    transition-duration: var(--terra-form-input-transition-duration, 0s);
    transition-property: background-size;
    transition-timing-function: var(--terra-form-input-transition-timing-function, ease);
    width: 100%;

    &:hover {
      border-color: var(--terra-form-input-hover-border-color, #4a4a4a);
      border-style: var(--terra-form-input-hover-border-style, solid);
      border-width: var(--terra-form-input-hover-border-width, 1px);
    }

    &[required] {
      box-shadow: unset;
    }

    &[disabled] {
      background-color: var(--terra-form-input-disabled-background-color, rgba(0, 0, 0, 0.05));
      border-color: var(--terra-form-input-disabled-border-color, #4a4a4a);
      border-style: var(--terra-form-input-disabled-border-style, solid);
      border-width: var(--terra-form-input-disabled-border-width, 1px);
      color: var(--terra-form-input-disabled-color, #000);
      cursor: default;
      opacity: var(--terra-form-input-disabled-opacity, 1);
    }

    &:focus {
      background-color: var(--terra-form-input-focus-background-color, #fff);
      border-color: var(--terra-form-input-focus-border-color, #4a4a4a);
      border-radius: var(--terra-form-input-focus-border-radius, 0.2rem);
      border-style: var(--terra-form-input-focus-border-style, solid);
      border-width: var(--terra-form-input-focus-border-width, 1px);
      opacity: var(--terra-form-input-focus-opacity, 1);
      outline: none;

      [dir=ltr] & {
        box-shadow: var(--terra-form-input-focus-keyboard-box-shadow-ltr, 0 0 8px #a6d9f4);
      }

      [dir=rtl] & {
        box-shadow: var(--terra-form-input-focus-keyboard-box-shadow-rtl, 0 0 8px #a6d9f4);
      }
    }
  }

  .form-error {
    background-image: var(--terra-form-input-error-focus-animation-background, none);
    border-color: var(--terra-form-input-error-border-color, #ff0b0f);
    border-radius: var(--terra-form-input-error-border-radius, 0.2rem);
    border-style: var(--terra-form-input-error-border-style, solid);
    border-width: var(--terra-form-input-error-border-width, 1px);
    opacity: var(--terra-form-input-error-opacity, 1);

    &:hover {
      border-color: var(--terra-form-input-error-hover-border-color, #ff0b0f);
      border-style: var(--terra-form-input-error-hover-border-style, solid);
      border-width: var(--terra-form-input-error-hover-border-width, 1px);
    }

    &[disabled] {
      background-color: var(--terra-form-input-error-disabled-background-color, rgba(0, 0, 0, 0.05));
      border-color: var(--terra-form-input-error-disabled-border-color, #ff0b0f);
      border-style: var(--terra-form-input-error-disabled-border-style, solid);
      border-width: var(--terra-form-input-error-disabled-border-width, 1px);
      color: var(--terra-form-input-error-disabled-color, #000);
      opacity: var(--terra-form-input-error-disabled-opacity, 1);
    }

    &:focus {
      background-color: var(--terra-form-input-error-focus-background-color, #fff);
      border-color: var(--terra-form-input-error-focus-border-color, #ff0b0f);
      border-radius: var(--terra-form-input-error-focus-border-radius, 0.2rem);
      border-style: var(--terra-form-input-error-focus-border-style, solid);
      border-width: var(--terra-form-input-error-focus-border-width, 1px);
      opacity: var(--terra-form-input-error-focus-opacity, 1);

      [dir=ltr] & {
        box-shadow: var(--terra-form-input-error-focus-keyboard-box-shadow-ltr, 0 0 8px #a6d9f4);
      }

      [dir=rtl] & {
        box-shadow: var(--terra-form-input-error-focus-keyboard-box-shadow-rtl, 0 0 8px #a6d9f4);
      }
    }
  }
}
