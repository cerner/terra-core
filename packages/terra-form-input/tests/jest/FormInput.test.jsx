import React from 'react';
import FormInput from '../../src/FormInput';

describe('FormInput', () => {
  const defaultRender = <FormInput />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.form-input').text()).toEqual('');
  });

  it('should render as uncontrolled when just a default value is passed into the FormInput', () => {
    const forminput = <FormInput defaultValue="foo" />;
    const wrapper = render(forminput);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as controlled when just a default value and `onChange()` is passed into the FormInput', () => {
    const forminput = <FormInput value="foo" onChange={() => {}} />;
    const wrapper = render(forminput);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set the forminput to disabled when passed into the component', () => {
    const forminput = <FormInput disabled />;
    const wrapper = render(forminput);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set the forminput to invalid when isInvalid is passed into the component', () => {
    const forminput = <FormInput isInvalid />;
    const wrapper = render(forminput);
    expect(wrapper).toMatchSnapshot();
  });

  // Structure Tests
  it('should have the class form-input', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('form-input');
  });
});
