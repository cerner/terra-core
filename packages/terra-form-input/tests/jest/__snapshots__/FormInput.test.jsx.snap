// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render a default component 1`] = `
<input
  aria-label="input label"
  className="form-input"
  disabled={false}
  name={null}
  required={false}
/>
`;

exports[`FormInput should render as controlled when just a default value and \`onChange()\` is passed into the FormInput 1`] = `
<input
  aria-label="input label"
  class="form-input"
  value="foo"
/>
`;

exports[`FormInput should render as uncontrolled when just a default value is passed into the FormInput 1`] = `
<input
  aria-label="input label"
  class="form-input"
  value="foo"
/>
`;

exports[`FormInput should set the forminput to disabled when passed into the component 1`] = `
<input
  aria-label="input label"
  class="form-input"
  disabled=""
/>
`;

exports[`FormInput should set the forminput to invalid when isInvalid is passed into the component 1`] = `
<input
  aria-label="input label"
  class="form-input form-error"
/>
`;
